ports/raspberrypi/bindings/rp2clock/InputPin.c

```diff
@@ -0,0 +1,149 @@
+        { MP_QSTR_pin,         MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_index,       MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_src_freq,    MP_ARG_REQUIRED | MP_ARG_INT },
+        { MP_QSTR_target_freq, MP_ARG_REQUIRED | MP_ARG_INT },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, all_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    rp2clock_inputpin_obj_t *self = m_new_obj_with_finaliser(rp2clock_inputpin_obj_t);
+    self->base.type = &rp2clock_inputpin_type;
+
+    // Validate pin number
+    common_hal_rp2clock_inputpin_validate_index_pin(args[ARG_pin].u_rom_obj);
+    self->pin = args[ARG_pin].u_rom_obj;
+
+    // Validate clock
+    self->index = cp_enum_value(&rp2clock_index_type, args[ARG_index].u_obj, MP_QSTR_rp2clock_index);
+
+    // Validate frequencies
+    common_hal_rp2clock_inputpin_validate_freqs(args[ARG_src_freq].u_int, args[ARG_target_freq].u_int);
+    self->src_freq = args[ARG_src_freq].u_int;
+    self->target_freq = args[ARG_target_freq].u_int;
+
+    // Enable and return
+    common_hal_rp2clock_inputpin_enable(self);
+    return MP_OBJ_FROM_PTR(self);
+};
+
+//|     def deinit(self) -> None:
+//|         """Releases the pin and frees any resources."""
+STATIC mp_obj_t rp2clock_inputpin_deinit(mp_obj_t self_in) {
+    rp2clock_inputpin_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    // Release pin
+    common_hal_rp2clock_inputpin_deinit(self);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(rp2clock_inputpin_deinit_obj, rp2clock_inputpin_deinit);
+
+//|     def enable(self) -> None:
+//|         """Configures the pin and enables the internal clock."""
+STATIC mp_obj_t rp2clock_inputpin_enable(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    rp2clock_inputpin_obj_t *self = MP_OBJ_TO_PTR(pos_args[0]);
+    check_for_deinit(self);
+    common_hal_rp2clock_inputpin_enable(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_KW(rp2clock_inputpin_enable_obj, 1, rp2clock_inputpin_enable);
+
+//|     def disable(self) -> None:
+//|         """Disables the pin and internal clock"""
+STATIC mp_obj_t rp2clock_inputpin_disable(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    rp2clock_inputpin_obj_t *self = MP_OBJ_TO_PTR(pos_args[0]);
+    check_for_deinit(self);
+    common_hal_rp2clock_inputpin_disable(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_KW(rp2clock_inputpin_disable_obj, 1, rp2clock_inputpin_disable);
+
+//|     def enabled(self) -> bool:
+//|         """Check if pin is enabled."""
+//|
+static mp_obj_t rp2clock_inputpin_get_enabled(mp_obj_t self_in) {
+    rp2clock_inputpin_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+    return mp_obj_new_bool(self->enabled);
+}
+MP_DEFINE_CONST_FUN_OBJ_1(rp2clock_inputpin_get_enabled_obj, rp2clock_inputpin_get_enabled);
+MP_PROPERTY_GETTER(rp2clock_inputpin_enabled_obj,
```

Comment:
Make this settable instead of having `enable()` and `disable()`.

https://github.com/adafruit/circuitpython/blob/8027efe24575fd4cfce0409c890348580ea291d2/shared-bindings/digitalio/DigitalInOut.c#L236-L260